### PR TITLE
Feature/browsetags

### PR DESCRIPTION
**Pull request by JorgenSolli**
referring to https://github.com/netgen/TagsBundle/pull/91

This is now a branch created directly from the '3.3-release'. Hopefully, everything is good now.
Please excuse me for taking this long to finally getting around to fixing this :D

### CODE REVIEW

**Comment by emodric** [comment]
Hi @JorgenSolli 

Thank you for your PR, it looks awesome!

I do, however, wonder, do we really need the possibility to customize the browsing templates? I'm referring to the changes in `bundle/Resources/views/ezadminui/field/edit/eztags.html.twig`. Or am I missunderstanding something?

Also, the `select.html.twig` file you modified belongs to the `Select` edit view (which renders dropdowns instead of autocomplete), so you will need to choose another template for your browse window. It is intentionally left blank because JavaScript takes care of showing the interface.

**Comment by JorgenSolli** [comment]
Hi @emodric 

You bring up a lot of valid points! I have tried to correct this.
Have a look and get back to me if more changes are required!

File: bundle/Resources/config/default_settings.yml [hunk 1]
```diff
@@ -34,6 +34,7 @@ parameters:
 
     # Setting that will hold all available edit views
     eztags.default.edit_views: {}
+    eztags.default.browse_views: {}
```
Comment (emodric) [inline bundle/Resources/config/default_settings.yml]: Not needed now, right?

File: bundle/Resources/views/ezadminui/field/edit/eztags.html.twig [hunk 2]
```diff
@@ -3,6 +3,7 @@
     {% set field_value = form.parent.vars.value.value %}
     {% set field_definition = form.parent.vars.value.fieldDefinition %}
     {% set edit_view = field_definition.fieldSettings.editView|default('Default') %}
+    {% set select_view = 'Select' %}
```
Comment (emodric) [inline bundle/Resources/views/ezadminui/field/edit/eztags.html.twig]: Leftover from previous version?

File: bundle/Resources/views/ezadminui/field/edit/eztags.html.twig [hunk 3]
```diff
@@ -27,13 +28,16 @@
...
-                    {% set edit_view_template = edit_view_config.template %}
+
+            {% for view_config in ezpublish.configResolver.parameter('edit_views', 'eztags') %}
+                {% if view_config.identifier == edit_view %}
+                    {% set edit_view_template = view_config.template %}
                 {% endif %}
             {% endfor %}
 
             {% include edit_view_template ignore missing %}
+            {% include '@NetgenTags/ezadminui/field/browse/view/browse.html.twig' ignore missing %}
```
Comment (emodric) [inline bundle/Resources/views/ezadminui/field/edit/eztags.html.twig]: No need for `ignore missing`, the template is obviously there now :)

Comment (emodric) [inline bundle/Resources/views/ezadminui/field/edit/eztags.html.twig]: No need for `ignore missing`, the template is obviously there now :)

File: bundle/Resources/views/ezadminui/field/edit/eztags.html.twig [hunk 4]
```diff
@@ -27,13 +28,16 @@
             </div>
 
             {% set edit_view_template = null %}
-            {% for edit_view_config in ezpublish.configResolver.parameter('edit_views', 'eztags') %}
-                {% if edit_view_config.identifier == edit_view %}
-                    {% set edit_view_template = edit_view_config.template %}
+
+            {% for view_config in ezpublish.configResolver.parameter('edit_views', 'eztags') %}
```
Comment (emodric) [inline bundle/Resources/views/ezadminui/field/edit/eztags.html.twig]: Can you bring back the old name `edit_view_config` now, to keep the diff smaller?

File: bundle/Resources/views/ezadminui/field/edit/view/select.html.twig [hunk 5]
```diff
@@ -1 +1,29 @@
 {# This file is intentionally left blank #}
+<div class="jqmDialog parent-selector-tree browse-tags" id="parent-selector-browse-tree-{{ field_definition.id }}">
```
Comment (emodric) [inline bundle/Resources/views/ezadminui/field/edit/view/select.html.twig]: These changes also need to reverted now.

**Comment by emodric** [comment]
@JorgenSolli Looks quite good now. I've left some more nitpicking comments before the merge.

**Comment by JorgenSolli** [comment]
@emodric Should be good to go now. Hopefully 🤞

**Comment by emodric** [comment]
Thanks :)

**Pull request merged**